### PR TITLE
Hash_spider Module

### PR DESCRIPTION
**Pull request by pgormanDS**
This module leverages BloodHound to spider across a domain from an initial account with local admin access.

hash_spider

### CODE REVIEW

**Comment by mpgn** [comment]
Nice, I will test it this week end :)

File: cme/modules/hash_spider.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+            context.log.error("Error querying domain admins")
+    else:
+        context.log.highlight("BloodHound not marked enabled. Check cme.conf")
+        sys.exit()
+
+def neo4j_local_admins(context):
+    global admin_results
+    try:
+        session = driver.session()
+        admins = session.run("MATCH (c:Computer) OPTIONAL MATCH (u1:User)-[:AdminTo]->(c) OPTIONAL MATCH (u2:User)-[:MemberOf*1..]->(:Group)-[:AdminTo]->(c) WITH COLLECT(u1) + COLLECT(u2) AS TempVar,c UNWIND TempVar AS Admins RETURN c.name AS COMPUTER, COUNT(DISTINCT(Admins)) AS ADMIN_COUNT,COLLECT(DISTINCT(Admins.name)) AS USERS ORDER BY ADMIN_COUNT DESC")
```
[the remark this answers is not in the document]
Comment (pgormanDS) [reply cme/modules/hash_spider.py:61]: Added a comment. The query pulls all PCs and their local admins from BloodHound so hash_spider knows what hashes to use on which machines.

File: cme/modules/hash_spider.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+found_users = []
+reported_da = []
+
+
+def neo4j_conn(context):
+    if config.get('BloodHound', 'bh_enabled') != "False":
+        context.log.info("Connecting to Neo4j/Bloodhound.")
+        try:
+            session = driver.session()
+            list(session.run("MATCH (g:Group) return g"))
```
[the remark this answers is not in the document]
Comment (pgormanDS) [reply cme/modules/hash_spider.py:45]: Lol, yeah that's smarter. Added limit.

**Comment by pgormanDS** [comment]
@mpgn All your comments should be resolved and fully functional again after I broke things. Please let me know if there's anything else I should fix or that I didn't fix properly.

**Comment by mpgn** [comment]
can you push the new code @pgormanDS  ? ^^

**Review by pgormanDS — COMMENTED** [review]
Cleaned up imports, moved db connection to on_admin_login, added limit to initial query check, added comment to Neo4j query.

**Comment by mpgn** [comment]
I update a little bit the code in case the user try with local admin :)

**Comment by mpgn** [comment]
@pgormanDS if you want your cme coin, pm on twitter (your dm are closed) 🥇

**Pull request merged**